Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs

```diff
@@ -178,33 +179,57 @@ public ISiteBuilder CreateBuilder(ITracer tracer, ILogger logger, IDeploymentSet
         {
+            string strFramework = System.Environment.GetEnvironmentVariable(OryxBuildConstants.OryxEnvVars.FrameworkSetting);
+            Framework framework = SupportedFrameworks.ParseLanguage(strFramework);
+
             string sourceProjectPath = projectPath ?? repositoryRoot;
-            if (FunctionAppHelper.LooksLikeFunctionApp())
-            {
-                return new FunctionBasicBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
-            }
-            else if (IsNodeSite(sourceProjectPath))
-            {
-                return new NodeSiteBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
-            }
-            else if (IsPythonSite(sourceProjectPath))
-            {
-                return new PythonSiteBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
-            }
-            else if (IsGoSite(sourceProjectPath))
+            switch (framework)
             {
-                return new GoSiteBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
-            }
-            else if (IsRubySite(sourceProjectPath))
-            {
-                return new RubySiteBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
-            }
-            else if (IsPHPSite(sourceProjectPath))
-            {
-                return new PHPSiteBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
-            }
+                case Framework.NodeJs:
+                    return new NodeSiteBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
+
+                case Framework.Python:
+                    return new PythonSiteBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
+
+                case Framework.Go:
+                    return new GoSiteBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
+
+                case Framework.Ruby:
+                    return new RubySiteBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
+                    
+                case Framework.PHP:
+                    return new PHPSiteBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
 
-            return new BasicBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
+                default:
+                    if (FunctionAppHelper.LooksLikeFunctionApp())
+                    {
+                        return new FunctionBasicBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
+                    }
+                    else if (IsRubySite(sourceProjectPath))
```

Comment:
- default to sniffers if linuxfxversion is not set, customer may be using a custom image, and relying on sniffers
- place ruby before node

Reply:
Before switching to other builders FunctionApp should still get the first precedence.

Reply:
Why do we have all the switch cases in default part again

Reply:
If we cant choose based on the linuxfxversion, we sniff the files